pkg/olog/logrus_test.go

```diff
@@ -0,0 +1,141 @@
+package olog
+
+import (
+	"bytes"
+	"errors"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sirupsen/logrus"
+	"gotest.tools/v3/assert"
+)
+
+func TestTextLogger(t *testing.T) {
+	buffer := bytes.Buffer{}
+	logger := logrus.New()
+	logger.SetFormatter(NewCharmTextFormatter())
+	logger.SetOutput(&buffer)
+	cases := []struct {
+		name     string
+		expected string
+		msg      string
+		fields   logrus.Fields
+		level    logrus.Level
+	}{
+		{
+			name:     "simple message",
+			expected: "INFO info\n",
+			msg:      "info",
+			fields:   nil,
+			level:    logrus.InfoLevel,
+		},
+		{
+			name:     "ignored message",
+			expected: "",
```

Comment:
Question: Why this debug log is expected to be ignored?

Reply:
This was copied over from the charm logger tests. I think it's because the default log level is Info, so debug logs are suppressed. This led me to realize that logrus puts the responsibility of suppressing logs on the formatter. I've added that, so it will now suppress logs below the logger's log level.